packages/@react-aria/tree/src/useTreeGridListItem.ts

```diff
@@ -36,6 +43,25 @@ export interface TreeGridListItemAria extends GridListItemAria {
  * @param ref - The ref attached to the row element.
  */
 export function useTreeGridListItem<T>(props: AriaTreeGridListItemOptions, state: TreeState<T>, ref: RefObject<FocusableElement>): TreeGridListItemAria {
-   // TODO: should it return a state specifically for isExpanded? Or is aria attribute sufficient?
-  return useGridListItem(props, state, ref);
+  let gridListAria = useGridListItem(props, state, ref);
+  let isExpanded = gridListAria.rowProps['aria-expanded'] === true;
+  let stringFormatter = useLocalizedStringFormatter(intlMessages, 'react-aria-components');
+
+  let expandButtonProps = {
+    onPress: () => {
+      if (!gridListAria.isDisabled) {
+        state.toggleKey(props.node.key);
+      }
+    },
+    'aria-label': isExpanded ? stringFormatter.format('collapse') : stringFormatter.format('expand'),
+    // TODO: the below actually isn't enough to have keyboard navigation skip over it, we need it to be a span type button but
```

Comment:
oh yeah, good point, now that we have that attribute I can hide the above from the tree walker I believe. Will add in followup

Reply:
went ahead and added it to the PR actually, got rid of the `element.getAttribute('aria-hidden') === 'true'` from the logic in `isAttributeVisible` since IMO our data attribute alone should be sufficient